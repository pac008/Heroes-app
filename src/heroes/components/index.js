export * from './HeroCard';
export * from './Herolist';