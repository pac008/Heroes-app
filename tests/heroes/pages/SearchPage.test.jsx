

import { fireEvent, render, screen } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"
import { Searchpage } from "../../../src/heroes"

const mockedUseNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUseNavigate
}))

// jest.mock('../../../src/heroes', () => ({
//     ...jest.requireActual('react-router-dom'),
//     // useNavigate: () => mockedUseNavigate
// }))

describe('test in SeatchPage', () => { 
    test('should display correctly with default values', () => { 
       const { container } = render(
            <MemoryRouter>
                <Searchpage />
            </MemoryRouter>
        )
        expect(container).toMatchSnapshot()
     });

    test('should display to batman and the Input with query values', () => { 
       render(
            <MemoryRouter initialEntries={['/search?q=batman']}>  
                <Searchpage />
            </MemoryRouter>
        );
        const input = screen.getByRole('textbox');
        expect(input.value).toBe('batman');
        const img = screen.getByRole('img');
        expect(img.src).toContain('/assets/heroes/dc-batman.jpg');
     });

    test('should display an error if the heroe is not founded (batman123)', () => { 
       render(
            <MemoryRouter initialEntries={['/search?q=batman123']}>  
                <Searchpage />
            </MemoryRouter>
        );
        const text1 = screen.getByText('Not hero with');
        const text2 = screen.getByText('batman123');
        expect(text1).toBeTruthy();
        expect(text2).toBeTruthy();
     });

    test('should call the navigate to the new screen', () => { 
       render(
            <MemoryRouter initialEntries={['/search?q=batman123']}>  
                <Searchpage />
            </MemoryRouter>
        );

        const input = screen.getByRole('textbox');
        fireEvent.change(input, {target: {value: 'superman'}})
        const form = screen.getByLabelText('form');
        fireEvent.submit(form);
        
        expect(mockedUseNavigate).toHaveBeenCalledWith('?q=superman')
     });

 })