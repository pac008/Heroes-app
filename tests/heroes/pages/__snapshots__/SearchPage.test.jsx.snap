// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in SeatchPage should display correctly with default values 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        aria-label="search-test"
        class="alert-primary alert animate__animated animate__fadeIn "
      >
         Search Hero
      </div>
    </div>
  </div>
</div>
`;
